src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -389,8 +389,8 @@ protected void pRequest2Txn(int type, long zxid, Request request, Record record,
                 parentPath = path.substring(0, lastSlash);
                 parentRecord = getRecordForPath(parentPath);
                 ChangeRecord nodeRecord = getRecordForPath(path);
-                checkACL(zks, parentRecord.acl, ZooDefs.Perms.DELETE,
-                        request.authInfo);
+                checkACL(zks, parentRecord.acl, ZooDefs.Perms.DELETE, request.authInfo);
```

Comment:
I see that when we create a znode we need to check the ACL of the parent. But do we still need to check the parent when we are deleting? /cc @phunt @fpj